themes/default/content/blog/iac-recommended-practices-tying-stacks-together/index.md

```diff
@@ -0,0 +1,89 @@
+
+## Linking stacks with stack references
+
+Stack references are a feature in Pulumi that allows you to access output values from one stack in another stack. In other words, stack references offer a way to programmatically "link" stacks in different projects for the purpose of sharing information. You can find more details on stack references [in the Pulumi docs](/docs/intro/concepts/stack/#stackreferences).
+
+Using stack references, the Zephyr team can make the necessary connections they needed:
+
+* The code handling the Kubernetes cluster can reference the VPC ID and subnet IDs from the base infrastructure layer.
+* Similarly, the code that is responsible for deploying the online store services can reference the necessary Kubernetes connection details (the Kubeconfig) for the Kubernetes cluster.
+
+It's worthwhile to note that any information that needs to be accessible from another stack via a stack reference must be exported as a stack output in the source stack. If you don't mark it as a stack output, then it can't be used in a stack reference. Adding stack outputs after the fact requires little effort and has no impact on existing infrastructure.
+
+Also, any value retrieved via a stack reference is treated as a Pulumi Output, and therefore may require some extra work to transform values (such as the use of `Output.apply`). The recent addition of `OutputDetails` support in Pulumi---you can read more about that [here](/blog/stack-reference-output-details/)---helps considerably in this situation. Some SDKs also have language-specific mechanisms that can help; for example, using Go's `.AsStringArrayOutput()` method on a `StackReference.GetOutput` statement makes referencing subnet IDs from other project much easier.
+
+While stack references are conceptually straightforward and not difficult to implement, there are some recommended practices to be mindful of regarding the use of stack outputs and stack references:
+
+1. When it comes to stack outputs, **export what is needed.** If it needs to be accessed from outside the stack, export it; otherwise, don't. It's simple to add more stack outputs after the fact, and as has been mentioned already this is done with a quick `pulumi up` that has no affect on existing infrastructure (you're modifying the stack object itself).
```

Comment:
>If it needs to be accessed from outside the stack, export it; otherwise, don't.

It occurred to me that after reading this, I might wonder, Why not? Is there any harm in doing so? There probably isn't, unless what you're exporting is somehow potentially sensitive of course. The "why" here might therefore be something along the lines of the same "why" that encourages encapsulation in a public API: expose what the API user cares about or is likely to need, hide all the rest as implementation detail (which ultimately leaves you with more ability to change the underlying implementation without affecting the API).

Reply:
I've added some text, borrowing from your suggestion, in 17aacd6f